lib/Mojo/BaseUtil.pm

```diff
@@ -0,0 +1,71 @@
+package Mojo::BaseUtil;
+use Mojo::Base -strict;
```

Comment:
This will either need to be replaced with the same pragmas as Mojo::Base uses, or Mojo::Base will need to require Mojo::BaseUtil as it did Mojo::Util previously, because of the circular require.

Reply:
Currently I used in Base `use Mojo::BaseUtil ();` and CI tests passed. IIUC that means we handle the circular require already. Or should I change it to require in Mojo::Base still? Can you elaborate?

Reply:
I think it makes sense for this to just `use strict; use warnings;` IMO. The whole point is to make it extremely clear that this package breaks circular deps.

Reply:
got it, done

Reply:
> Currently I used in Base `use Mojo::BaseUtil ();` and CI tests passed. IIUC that means we handle the circular require already. Or should I change it to require in Mojo::Base still? Can you elaborate?

This will still fail to use strict and warnings and the other pragmas which -strict is supposed to import, because of the circular require. Specifically, Mojo::BaseUtil is loaded during the compilation phase of Mojo::Base, and attempts to load Mojo::Base which has not finished loading and so silently ignores its import method.